### DISCUSSION TITLE
how to map key like <c-w> <c-s-w> <c-w:i> <c-w:o>

### DISCUSSION THREAD

**Discussion by phanen** [gdh1995/vimium-c#926] [Q&A] [answered]
## what I want
I want to map `<c-w:i>` `<c-w:o>` in insert/vomnibar mode to delete a word, but it didn't work:
```
map <c-w:i> editText run="move,backward,word,exec,delete,"
map <c-w:o> editText run="move,backward,word,exec,delete,"
```
When press `<c-w>` while editing, it simply closes the window...

### map like `<c-w>` `<c-t>` doesn't work
I try to config in custom key mappings:
```
map <c-w> scrollDown
map <c-t> scrollDown
map <c-n> scrollDown
map <c-s-w> scrollDown
map <c-s-t> scrollDown
map <c-s-n> scrollDown
```
However, those keys still works as default behaviors in chromium(close tab, new tab, new window...).
But any other keys except `w`, `t`, `n` can be mapped and work well.

### map like `<c-w:i>` doesn't work
```
map <c-xx:i> editText run="move,backward,word,exec,delete,"
```
Similarly, `<c-xx:i>` seems work in insert mode, as long as `xx` is not `w`, `t`, `n`...

### map like `<c-xx:o>` doesn't work
And I try to map keys in vomnibar like [wiki](https://github.com/gdh1995/vimium-c/wiki/Use-in-another-keyboard-layout#per-mode-mapkey).
```
map <c-xx:o> editText run="move,backward,word,exec,delete,"
```
It seems that any key `xx` can not be mapped.

## environment
- Browser name: chromium
- Browser version: 112.0.5615.121 (Official Build) Arch Linux (64-bit) 
- Vimium C version: 1.99.98
- OS name and version: Linux arch 6.2.11-arch1-1

**Comment by gdh1995** [comment, marked as answer]
impossible, limited by browser



---Original---
From: "Xiaohan ***@***.***&gt;
Date: Mon, Apr 17, 2023 11:17 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [gdh1995/vimium-c] how to map key like <c-w&gt; <c-s-w&gt; <c-w:i&gt; <c-w:o&gt;(Discussion #926)




 
what I want
 
I want to map <c-w:i&gt; <c-w:o&gt; in insert/vomnibar mode to delete a word, but it didn't work:
 map <c-w:i&gt; editText run="move,backward,word,exec,delete," map <c-w:o&gt; editText run="move,backward,word,exec,delete,"  
When press <c-w&gt; while editing, it simply closes the window...
 
I have also try...
 
map like <c-w&gt; <c-t&gt; doesn't work
 
I try to config in custom key mappings:
 map <c-w&gt; scrollDown map <c-t&gt; scrollDown map <c-n&gt; scrollDown map <c-s-w&gt; scrollDown map <c-s-t&gt; scrollDown map <c-s-n&gt; scrollDown  
However, those keys still works as default behaviors in chromium(close tab, new tab, new window...).
 But any other keys except w, t, n can be mapped and work well.
 
map like <c-w:i&gt; doesn't work
 map <c-xx:i&gt; editText run="move,backward,word,exec,delete,"  
Similarly, <c-xx:i&gt; seems work in insert mode, as long as xx is not w, t, n...
 
map like <c-w:i&gt; <c-t&gt; doesn't work
 
And I try to map key in vomnibar like (wiki)[https://github.com/gdh1995/vimium-c/wiki/Use-in-another-keyboard-layout#per-mode-mapkey].
 map <c-&gt;  
environment
  
Browser name: chromium
 
Browser version: 112.0.5615.121 (Official Build) Arch Linux (64-bit)
 
Vimium C version: 1.99.98
 
OS name and version: Linux arch 6.2.11-arch1-1
